susemanager-utils/susemanager-sls/salt/bootstrap/init.sls

```diff
@@ -23,15 +23,26 @@ disable_repo_{{ alias }}:
 {%- endif %}
 {%- endfor %}
 
+{% set os_base = 'sle' %}
+{%- if "centos" in grains['oscodename']|lower %}
+{% set os_base = 'centos' %}
+{%- elif "opensuse" in grains['oscodename']|lower %}
+{% set os_base = 'opensuse' %}
+{%- endif %}
+
 {%- if grains['os_family'] == 'Suse' %}
 {%- if "." in grains['osrelease'] %}
-{% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/sle/' ~ grains['osrelease'].replace('.', '/') ~ '/bootstrap/' %}
+{% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/' ~ os_base ~ '/' ~ grains['osrelease'].replace('.', '/') ~ '/bootstrap/' %}
 {%- else %}
-{% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/sle/' ~ grains['osrelease'] ~ '/0/bootstrap/' %}
+{% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/' ~ os_base ~ '/' ~ grains['osrelease'] ~ '/0/bootstrap/' %}
 {%- endif %}
 {%- elif grains['os_family'] == 'RedHat' %}
+{% if salt['file.directory_exists' ]('/etc/centos-release') %}
+{% set bootstrap_repo_url = 'https://' ~ salt['pillar.get']('mgr_server') ~ '/pub/repositories/' ~ os_base ~ '/' ~ grains['osmajorrelease'] ~ '/bootstrap/' %}
```

Comment:
@chiaradiamarcelo, it seems to me that there is something that is not working with `os_base`.

When booting a CentOS6 client, we see at the logs:
```
2018-12-07 14:30:00,624 [salt.utils.http  ][DEBUG   ][5183] Requesting URL https://client-test-server.mgr.suse.de/pub/repositories/res/6/bootstrap/repodata/repomd.xml using GET method
```
But of course, the bootstrap repo for CentOS6 is at https://client-test-server.mgr.suse.de/pub/repositories/centos/6/bootstrap/repodata/.

I checked, and yes, file exists at `/etc/centos-release`.

@mcalmer is now talking to @isbm to see if with `salt-ssh` the jinja evaluation happens on the client as well.

Reply:
```
{% if salt['file.directory_exists' ]('/etc/centos-release') %}
```
LOL

It is `file.file_exists`, guys...